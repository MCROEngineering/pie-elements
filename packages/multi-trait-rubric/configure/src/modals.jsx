import React from 'react';
import PropTypes from 'prop-types';

import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import Button from '@material-ui/core/Button';

export const excludeZeroTypes = {
  remove0: 'remove0',
  add0: 'add0',
  shiftLeft: 'shiftLeft',
  shiftRight: 'shiftRight'
};

export const ExcludeZeroDialog = ({ open, changeExcludeZero, cancel }) => (
  <Dialog open={open}>
    <DialogTitle>
      Exclude 0 (Zero) from Score Point Values.
    </DialogTitle>

    <DialogContent>
      <DialogContentText>
        You are about to exclude 0 from score point values.
        <br/>
        Some of the existing data has to be changed.
        <br/>
        Please choose if you want to:
        <ul>
          <li>
            shift Labels and Descriptions to the left
          </li>
          <li>
            remove 0 column with its Label and Description
          </li>
        </ul>
      </DialogContentText>
    </DialogContent>

    <DialogActions>
      <Button onClick={() => changeExcludeZero(excludeZeroTypes.shiftLeft)}>
        Shift to Left
      </Button>

      <Button onClick={() => changeExcludeZero(excludeZeroTypes.remove0)}>
        Remove 0 column
      </Button>

      <Button onClick={cancel}>
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

ExcludeZeroDialog.propTypes = {
  open: PropTypes.bool,
  changeExcludeZero: PropTypes.func,
  cancel: PropTypes.func,
};

export const IncludeZeroDialog = ({ open, changeExcludeZero, cancel }) => (
  <Dialog open={open}>
    <DialogTitle>
      Include 0 (Zero) in Score Point Values.
    </DialogTitle>

    <DialogContent>
      <DialogContentText>
        You are about to include 0 in score point values.
        <br/>
        Some of the existing data has to be changed.
        <br/>
        Please choose if you want to:
        <ul>
          <li>
            shift Labels and Descriptions to the right
          </li>
          <li>
            add 0 column with empty Label and Descriptions
          </li>
        </ul>
      </DialogContentText>
    </DialogContent>

    <DialogActions>
      <Button onClick={() => changeExcludeZero(excludeZeroTypes.shiftRight)}>
        Shift to Right
      </Button>

      <Button onClick={() => changeExcludeZero(excludeZeroTypes.add0)}>
        Add 0 column
      </Button>

      <Button onClick={cancel}>
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

IncludeZeroDialog.propTypes = {
  open: PropTypes.bool,
  changeExcludeZero: PropTypes.func,
  cancel: PropTypes.func,
};

export const DecreaseMaxPoints = ({ open, deleteScorePoints, cancel }) => (
  <Dialog open={open}>
    <DialogTitle>
      Decreasing Max Points.
    </DialogTitle>

    <DialogContent>
      <DialogContentText>
        You are about to decrease max score point value.
        <br/>
        All the Labels and Descriptions for scores above Max Point will be deleted.
      </DialogContentText>
    </DialogContent>

    <DialogActions>
      <Button onClick={deleteScorePoints}>
        Ok, delete.
      </Button>

      <Button onClick={cancel}>
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

DecreaseMaxPoints.propTypes = {
  open: PropTypes.bool,
  deleteScorePoints: PropTypes.func,
  cancel: PropTypes.func,
};

export const DeleteScale = ({ open, scaleIndex, deleteScale, cancel }) => (
  <Dialog open={open}>
    <DialogTitle>
      Delete Scale #{scaleIndex}
    </DialogTitle>

    <DialogContent>
      <DialogContentText>
        Are you sure you want to delete this scale?
      </DialogContentText>
    </DialogContent>

    <DialogActions>
      <Button onClick={deleteScale}>
        Delete
      </Button>

      <Button onClick={cancel}>
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

DeleteScale.propTypes = {
  open: PropTypes.bool,
  scaleIndex: PropTypes.number,
  deleteScale: PropTypes.func,
  cancel: PropTypes.func,
};

export const DeleteTrait = ({ open, deleteTrait, cancel }) => (
  <Dialog open={open}>
    <DialogTitle>
      Delete Trait
    </DialogTitle>

    <DialogContent>
      <DialogContentText>
        Are you sure you want to delete this trait?
      </DialogContentText>
    </DialogContent>

    <DialogActions>
      <Button onClick={deleteTrait}>
        Delete
      </Button>

      <Button onClick={cancel}>
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

DeleteTrait.propTypes = {
  open: PropTypes.bool,
  deleteTrait: PropTypes.func,
  cancel: PropTypes.func,
};
