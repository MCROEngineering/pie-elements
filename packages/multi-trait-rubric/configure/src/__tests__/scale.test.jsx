import { shallow } from 'enzyme';
import React from 'react';

import { Scale } from '../scale';
import { excludeZeroTypes } from '../modals';

const scale = () => ({
  excludeZero: false,
  maxPoints: 2,
  scorePointsLabels: ['Non-Scorable', 'Unsatisfactory', 'Satisfactory'],
  traitLabel: 'Category',
  traits: [
    {
      name: 'Presentation',
      standards: [],
      description: '',
      scorePointsDescriptors: [
        'Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ',
        'Handwriting poor\n' + '\n' + 'Overall appearance is distracting to unacceptable',
        'Handwriting is generally legible\n' + '\n' + 'Overall appearance is acceptable or better',
      ],
    },
  ]
});

describe('Scale', () => {
  let w;
  let onScaleChanged = jest.fn();
  let onScaleRemoved = jest.fn();

  const wrapper = extras => {
    const defaults = {
      classes: {},
      scale: scale(),
      scaleIndex: 0,
      onScaleChanged,
      onScaleRemoved,
      showStandards: true,
      ...extras
    };

    return shallow(<Scale {...defaults} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });

    it('renders without traits', () => {
      w = wrapper({ traits: [] });
      expect(w).toMatchSnapshot();
    });

    it('renders without standards', () => {
      w = wrapper({ showStandards: false });
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    beforeEach(() => {
      w = wrapper();
    });

    describe('changeExcludeZero', () => {
      it('removes zero', () => {
        const { traits, scorePointsLabels, excludeZero } = w.instance().props.scale;
        w.instance().changeExcludeZero(excludeZeroTypes.remove0);

        expect(onScaleChanged).toBeCalledWith(0, {
          excludeZero: !excludeZero,
          scorePointsLabels: scorePointsLabels.slice(1),
          traits: [{
            ...traits[0],
            scorePointsDescriptors: traits[0].scorePointsDescriptors.slice(1)
          }]
        });
      });
      it('add0 zero', () => {
        const { traits, scorePointsLabels, excludeZero } = w.instance().props.scale;
        w.instance().changeExcludeZero(excludeZeroTypes.add0);

        expect(onScaleChanged).toBeCalledWith(0, {
          excludeZero: !excludeZero,
          scorePointsLabels: ['', ...scorePointsLabels],
          traits: [{
            ...traits[0],
            scorePointsDescriptors: ['', ...traits[0].scorePointsDescriptors]
          }]
        });
      });
      it('shift to Left', () => {
        const { traits, scorePointsLabels, excludeZero } = w.instance().props.scale;
        w.instance().changeExcludeZero(excludeZeroTypes.shiftLeft);

        expect(onScaleChanged).toBeCalledWith(0, {
          excludeZero: !excludeZero,
          scorePointsLabels: scorePointsLabels.slice(0, -1),
          traits: [{
            ...traits[0],
            scorePointsDescriptors: traits[0].scorePointsDescriptors.slice(0, -1)
          }]
        });
      });
      it('shift to Right', () => {
        const { traits, scorePointsLabels, excludeZero } = w.instance().props.scale;
        w.instance().changeExcludeZero(excludeZeroTypes.shiftRight);

        expect(onScaleChanged).toBeCalledWith(0, {
          excludeZero: !excludeZero,
          scorePointsLabels: [...scorePointsLabels, ''],
          traits: [{
            ...traits[0],
            scorePointsDescriptors: [...traits[0].scorePointsDescriptors, '']
          }
          ]
        })
        ;
      });
    });

    describe('updateMaxPointsFieldValue', () => {
      it('shows alert box is number less then max points', () => {
        const { maxPoints } = w.instance().props.scale;
        w.instance().updateMaxPointsFieldValue({ target: { value: maxPoints - 1 } });

        expect(w.instance().state.newMaxPoints).toEqual(maxPoints - 1);
      });

      it('changes max points if number more then max points', () => {
        const { maxPoints } = w.instance().props.scale;
        w.instance().updateMaxPointsFieldValue({ target: { value: maxPoints + 1 } });

        expect(onScaleChanged).toBeCalledWith(0, { maxPoints: maxPoints + 1 });
      });
    });

    describe('changeMaxPoints', () => {
      it('removes zero', () => {
        w.instance().setState({ newMaxPoints: 10 });
        w.instance().changeMaxPoints();

        expect(onScaleChanged).toBeCalledWith(0, { maxPoints: 10 });
      });
    });

    describe('deleteScale', () => {
      it('calls onRemoveScale', () => {
        w.instance().deleteScale();

        expect(onScaleRemoved).toBeCalledWith(0);
      });
    });

    describe.only('onTraitRemoved', () => {
      // todo
      // it('calls onScaleChanged', () => {
      //   w.instance().setState({ traitToDeleteIndex: 0 });
      //   w.instance().onTraitRemoved();
      //
      //   expect(onScaleRemoved).toBeCalledWith(0, { traits: [] });
      // });

      it('does not call onScaleChanged if index not valid', () => {
        w.instance().setState({ traitToDeleteIndex: -1 });
        const result = w.instance().onTraitRemoved();

        expect(result).toEqual(false);
      });
    });

    describe.only('onTraitChanged', () => {
      it('calls onScaleChanged', () => {
        w.instance().onTraitChanged(0, {});

        expect(onScaleChanged).toBeCalledWith(0, { traits: [{}] });
      });
    });

    // todo
    // describe.only('onTraitDropped', () => {
    //   it('calls onScaleChanged', () => {
    //     w.instance().onTraitDropped({ index: 0 }, 1);
    //
    //     expect(onScaleChanged).toBeCalledWith(0, { traits: [{}]});
    //   });
    // });
  });
});
