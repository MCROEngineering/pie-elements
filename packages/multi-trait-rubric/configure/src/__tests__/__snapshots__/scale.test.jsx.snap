// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale snapshot renders 1`] = `
<div
  key="scale-0"
>
  <div>
    <h3
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Scale #
      0
    </h3>
    <pure(DeleteIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={
      <WithStyles(Checkbox)
        checked={false}
        color="primary"
        onChange={[Function]}
      />
    }
    label="Exclude Zero"
    value="exclude_zero"
  />
  <WithStyles(FormControl)>
    <WithStyles(WithFormControlContext(InputLabel))>
      Max Points
    </WithStyles(WithFormControlContext(InputLabel))>
    <WithStyles(WithFormControlContext(Select))
      input={<WithStyles(WithStyles(WithFormControlContext(InputBase))) />}
      onChange={[Function]}
      value={2}
    >
      <WithStyles(MenuItem)
        key="menu-item-1"
        value={1}
      >
        1
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-2"
        value={2}
      >
        2
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-3"
        value={3}
      >
        3
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-4"
        value={4}
      >
        4
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-5"
        value={5}
      >
        5
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-6"
        value={6}
      >
        6
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-7"
        value={7}
      >
        7
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-8"
        value={8}
      >
        8
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-9"
        value={9}
      >
        9
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-10"
        value={10}
      >
        10
      </WithStyles(MenuItem)>
    </WithStyles(WithFormControlContext(Select))>
  </WithStyles(FormControl)>
  <WithStyles(TraitsHeaderTile)
    key="header-key"
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={true}
    traitLabel="Category"
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <div>
    <div>
      <div>
        Add
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Category",
          }
        }
      />
    </div>
    <pure(AddCircleIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <ExcludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <IncludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <DecreaseMaxPoints
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <DeleteScale
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <DeleteTrait
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders without standards 1`] = `
<div
  key="scale-0"
>
  <div>
    <h3
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Scale #
      0
    </h3>
    <pure(DeleteIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={
      <WithStyles(Checkbox)
        checked={false}
        color="primary"
        onChange={[Function]}
      />
    }
    label="Exclude Zero"
    value="exclude_zero"
  />
  <WithStyles(FormControl)>
    <WithStyles(WithFormControlContext(InputLabel))>
      Max Points
    </WithStyles(WithFormControlContext(InputLabel))>
    <WithStyles(WithFormControlContext(Select))
      input={<WithStyles(WithStyles(WithFormControlContext(InputBase))) />}
      onChange={[Function]}
      value={2}
    >
      <WithStyles(MenuItem)
        key="menu-item-1"
        value={1}
      >
        1
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-2"
        value={2}
      >
        2
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-3"
        value={3}
      >
        3
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-4"
        value={4}
      >
        4
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-5"
        value={5}
      >
        5
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-6"
        value={6}
      >
        6
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-7"
        value={7}
      >
        7
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-8"
        value={8}
      >
        8
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-9"
        value={9}
      >
        9
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-10"
        value={10}
      >
        10
      </WithStyles(MenuItem)>
    </WithStyles(WithFormControlContext(Select))>
  </WithStyles(FormControl)>
  <WithStyles(TraitsHeaderTile)
    key="header-key"
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={false}
    traitLabel="Category"
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={false}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <div>
    <div>
      <div>
        Add
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Category",
          }
        }
      />
    </div>
    <pure(AddCircleIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <ExcludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <IncludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <DecreaseMaxPoints
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <DeleteScale
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <DeleteTrait
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders without traits 1`] = `
<div
  key="scale-0"
>
  <div>
    <h3
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Scale #
      0
    </h3>
    <pure(DeleteIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={
      <WithStyles(Checkbox)
        checked={false}
        color="primary"
        onChange={[Function]}
      />
    }
    label="Exclude Zero"
    value="exclude_zero"
  />
  <WithStyles(FormControl)>
    <WithStyles(WithFormControlContext(InputLabel))>
      Max Points
    </WithStyles(WithFormControlContext(InputLabel))>
    <WithStyles(WithFormControlContext(Select))
      input={<WithStyles(WithStyles(WithFormControlContext(InputBase))) />}
      onChange={[Function]}
      value={2}
    >
      <WithStyles(MenuItem)
        key="menu-item-1"
        value={1}
      >
        1
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-2"
        value={2}
      >
        2
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-3"
        value={3}
      >
        3
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-4"
        value={4}
      >
        4
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-5"
        value={5}
      >
        5
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-6"
        value={6}
      >
        6
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-7"
        value={7}
      >
        7
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-8"
        value={8}
      >
        8
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-9"
        value={9}
      >
        9
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="menu-item-10"
        value={10}
      >
        10
      </WithStyles(MenuItem)>
    </WithStyles(WithFormControlContext(Select))>
  </WithStyles(FormControl)>
  <WithStyles(TraitsHeaderTile)
    key="header-key"
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={true}
    traitLabel="Category"
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={
      Array [
        2,
        1,
        0,
      ]
    }
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <div>
    <div>
      <div>
        Add
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Category",
          }
        }
      />
    </div>
    <pure(AddCircleIcon)
      classes={
        Object {
          "root": undefined,
        }
      }
      onClick={[Function]}
    />
  </div>
  <ExcludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <IncludeZeroDialog
    cancel={[Function]}
    changeExcludeZero={[Function]}
    open={false}
  />
  <DecreaseMaxPoints
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <DeleteScale
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <DeleteTrait
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;
